routes/teams.js

```diff
@@ -34,10 +34,107 @@ router.get('/', function(res, res) {
 router.get('/team', function(req, res) {
     var term = req.query.q;
     console.log(term);
-    //find the team, This syntax does a sql-type like clause with case insensitivy(sp???) with the RegEx
+    //find the team, This syntax does a sql-type like clause with case insensitivity with the RegEx
     db.collection('teams').find({$or: [{'name': new RegExp(term, 'i')}, {'market': new RegExp(term, 'i')}]}).toArray(function (err, items) {
     	res.json(items);
     });
 });
 
+
+// when players endpoint is hit, call the API/DB using that team_id
+router.get('/players', function(req, res) {
+  team_id = req.query["team_id"];
+  players = fetchPlayers(team_id, res, returnPlayers);
+});
+
+var returnPlayers = function (players, res){
+    res.json(players);
+}
+
+  
+var fetchPlayers = function(team_id, res, callback){
+  db.collection('players').find({team_id : team_id}).toArray(function (err, items){
+    if (items.length > 0){ // data in Mongo
+      var date = items[0]["last_updated"].valueOf();
+      var datenow = new Date();
+      var datecutoff = datenow.getTime() - 86400000;
+      if (datecutoff > date){   //data is old so call API
+        var players = fetchPlayersFromApi(team_id,res,callback)
+      }
+      else {  // data is fine so just return it
+        callback(items, res);
+      }
+    }
+    else {  // data not already in Mongo
+      var players = fetchPlayersFromApi(team_id,res,callback)
+    }
+  });
+}
+
+
+
+var fetchPlayersFromApi = function(team_id,res,callback){
+var json_response = '';
+var players = {};
+  request('https://api.sportsdatallc.org/nba-t3/seasontd/2014/reg/teams/'+team_id+'/statistics.json?api_key='+nba_key, function (error, response, body) {
+    if (!error && response.statusCode == 200) {
+      json_response = JSON.parse(body);
+      players = formatPlayers(json_response, team_id);
+      mongoInsertPlayers(team_id, players);
+      callback(players, res)
+    } 
+  }); 
+}
+
+
+var formatPlayers = function(response, team_id){
+  playersarray = [];
+  for (i=0;i<response.players.length;i++){
+    playersarray[i] = {};
+    for(var key in response.players[i]){
+      if (key != "total" && key != "average"){
+        var value = response.players[i][key];
+        playersarray[i][key] = value;
+      }
+      else if (key =="total"){
+        for(var key in response.players[i].total) {
+          var value = response.players[i].total[key];
+          playersarray[i]["total_"+key] = value;
```

Comment:
Not quite. API serves this up as a totals object within the players array that has "three_pt_attmpts", "two_pt_attmps" inside. I've flattened it so there isn't a totals object and an average object.. but all those key/vals are in the main players array properly labeled.